Fix IVY-1528

jaikiran opened the pull request:
The commit here fixes the issue reported in https://issues.apache.org/jira/browse/IVY-1528 and includes a test case to verify the fix. With this fix, the `classifier` extra attribute is no longer ignored while generating the `pom` file through the `makepom` task.

janmaterne commented:
- we dont use @author
- TestXmlHelper
  -- if xPathExpression would be a String and the method compiles that for itself,
     reuse would be easier
  -- XmlHelper would be a better name, because does not work only on tests.
  -- result could be inlined
- IvyMakePomTest#testClassifier()
  -- you dont use expectedPomArtifactIds for assertions. Do you want to check
     emptyness after the loop?

jaikiran commented:
>> we dont use @author

Fixed - removed that tag. I will setup my IDE accordingly.

>> TestXmlHelper
-- if xPathExpression would be a String and the method compiles that for itself,
reuse would be easier

Done

>> -- XmlHelper would be a better name, because does not work only on tests.

For now I wanted that class to stay just for the tests usage, since it's currently not used/required in other places. I can move it to the non-test source package if you prefer and rename it to XmlHelper.

>> -- result could be inlined

Done

>> IvyMakePomTest#testClassifier()
-- you dont use expectedPomArtifactIds for assertions. Do you want to check
emptyness after the loop?

Good catch. Fixed.

The PR has been updated with these changes. Let me know if there's any other changes that you recommend.

janmaterne commented:
It's ok. What's not ok is my own environmen: I can see your changes on PomModuleDescriptorWriter.java, but I am missing all new files.
I used 
git fetch github pull/34/head:PR-34
git checkout PR-34

Some ideas?

jaikiran commented:
I used
>> git fetch github pull/34/head:PR-34
>> git checkout PR-34

>> Some ideas?

What does the output of:

```git status```

and then

```git show```
output locally?

janmaterne commented:
Thanks. It was the most common error: the user error :)
Used the GUI "git history" not on the top folder ...
